### ISSUE TITLE
Old CocoaLumberjack version

### ISSUE THREAD

**Issue by KowalskiP** [expo/expo#4481]
## 🐛 Bug Report

I have a package which depends on new version of CocoaLumberjack (~> 3.5.1). This package is important for project, so I can't remove it. Also I can't control version of it's CocoaLumberjack. Right now ExpoKit/Core depends on 3.2.1 version.
Is there a way to solve this problem.

### Environment

Expo CLI 2.19.2 environment info:
    System:
      OS: macOS 10.14.5
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 8.12.0 - ~/.nvm/versions/node/v8.12.0/bin/node
      Yarn: 1.16.0 - /usr/local/bin/yarn
      npm: 6.5.0 - ~/.nvm/versions/node/v8.12.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      @types/expo: ^32.0.13 => 32.0.13
      @types/react: ^16.8.19 => 16.8.19
      @types/react-native: ^0.57.60 => 0.57.60
      @types/react-navigation: ^3.0.7 => 3.0.7
      expo: ^33.0.0 => 33.0.3
      react: 16.8.3 => 16.8.3
      react-native: https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz => 0.59.8
      react-navigation: ^3.11.0 => 3.11.0
    npmGlobalPackages:
      expo-cli: 2.19.2

ExpoKit, iOS

### Steps to Reproduce

```sh
cd ios && pod update
```

### Expected Behavior

Pods should be installed correctly.

### Actual Behavior
Pods didn't installed

**Comment by cruzach** [comment]
Hi @KowalskiP- What version of `cocoapods` are you using?

**Comment by KowalskiP** [comment]
Hi @cruzach 
```sh
pod --version
1.5.3
```

**Comment by cruzach** [comment]
And what error is getting printed out when you try to run `pod install`?

**Comment by KowalskiP** [comment]
```sh
[!] CocoaPods could not find compatible versions for pod "CocoaLumberjack":
  In Podfile:
    ExpoKit/Core (from `http://github.com/expo/expo.git`, tag `ios/2.11.1`) was resolved to 33.0.0, which depends on
      CocoaLumberjack (~> 3.2.1)

    RNBackgroundGeolocation (from `../node_modules/react-native-background-geolocation`) was resolved to 3.0.6, which depends on
      CocoaLumberjack (~> 3.5.1)
```

**Comment by KowalskiP** [comment]
Tried with version 1.7.1 - same error

**Comment by KowalskiP** [comment]
Podfile:
```rb
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '10.0'

target 'project' do
  pod 'ExpoKit',
    :git => "http://github.com/expo/expo.git",
    :tag => "ios/2.11.1",
    :subspecs => [
      "Core"
    ],
    :inhibit_warnings => true

  # Install unimodules
  require_relative '../node_modules/react-native-unimodules/cocoapods.rb'
  use_unimodules!

  pod 'React',
    :path => "../node_modules/react-native",
    :inhibit_warnings => true,
    :subspecs => [
      "Core",
      "ART",
      "RCTActionSheet",
      "RCTAnimation",
      "RCTCameraRoll",
      "RCTGeolocation",
      "RCTImage",
      "RCTNetwork",
      "RCTText",
      "RCTVibration",
      "RCTWebSocket",
      "DevSupport",
      "CxxBridge"
    ]
  pod 'yoga',
    :path => "../node_modules/react-native/ReactCommon/yoga",
    :inhibit_warnings => true
  pod 'DoubleConversion',
    :podspec => "../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec",
    :inhibit_warnings => true
  pod 'Folly',
    :podspec => "../node_modules/react-native/third-party-podspecs/Folly.podspec",
    :inhibit_warnings => true
  pod 'glog',
    :podspec => "../node_modules/react-native/third-party-podspecs/glog.podspec",
    :inhibit_warnings => true

  pod 'RNBackgroundGeolocation', :path => '../node_modules/react-native-background-geolocation'

  pod 'ReactNativeFabric', :path => '../node_modules/react-native-fabric'

  pod 'RNBackgroundFetch', :path => '../node_modules/react-native-background-fetch'

  pod 'Firebase/Core', '~> 5.15.0'
  pod 'Firebase/Messaging', '~> 5.15.0'

  pod 'SentryReactNative', :path => '../node_modules/react-native-sentry'

  pod 'react-native-config', :path => '../node_modules/react-native-config'

  pod 'react-native-background-timer', :path => '../node_modules/react-native-background-timer'


  post_install do |installer|
    installer.pods_project.main_group.tab_width = '2';
    installer.pods_project.main_group.indent_width = '2';

    installer.target_installation_results.pod_target_installation_results
      .each do |pod_name, target_installation_result|

      if pod_name == 'ExpoKit'
        target_installation_result.native_target.build_configurations.each do |config|
          config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)']
          config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << 'EX_DETACHED=1'
          
          # Enable Google Maps support
          config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << 'HAVE_GOOGLE_MAPS=1'
          config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << 'HAVE_GOOGLE_MAPS_UTILS=1'
          
        end
      end


      if ['Amplitude-iOS','Analytics','AppAuth','Branch','CocoaLumberjack','FBSDKCoreKit','FBSDKLoginKit','FBSDKShareKit','GPUImage','JKBigInteger2'].include? pod_name
        target_installation_result.native_target.build_configurations.each do |config|
          config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '10.0'
        end
      end

      # Can't specify this in the React podspec because we need to use those podspecs for detached
      # projects which don't reference ExponentCPP.
      if pod_name.start_with?('React')
        target_installation_result.native_target.build_configurations.each do |config|
          config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '10.0'
          config.build_settings['HEADER_SEARCH_PATHS'] ||= ['$(inherited)']
        end
      end

      # Build React Native with RCT_DEV enabled and RCT_ENABLE_INSPECTOR and
      # RCT_ENABLE_PACKAGER_CONNECTION disabled
      next unless pod_name == 'React'
      target_installation_result.native_target.build_configurations.each do |config|
        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)']
        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << 'RCT_DEV=1'
        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << 'RCT_ENABLE_INSPECTOR=0'
        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] << 'ENABLE_PACKAGER_CONNECTION=0'
      end
      if target.pod_name == 'react-native-config'
        phase = target.project.new(Xcodeproj::Project::Object::PBXShellScriptBuildPhase)
        phase.shell_script = "cd ../../"\
                             " && RNC_ROOT=./node_modules/react-native-config/"\
                             " && export SYMROOT=$RNC_ROOT/ios/ReactNativeConfig"\
                             " && export BUILD_DIR=$RNC_ROOT/ios/ReactNativeConfig"\
                             " && ruby $RNC_ROOT/ios/ReactNativeConfig/BuildDotenvConfig.ruby"
  
        target.build_phases << phase
        target.build_phases.move(phase,0)
      end
    end
  end
end

```

**Comment by KowalskiP** [comment]
Forked your repo and update cocoalumberjack dependency solve this problem. I hope expo will update this dependency soon because using a fork is not good idea for maintaining

**Comment by cruzach** [comment]
@KowalskiP - you updated to 3.5.1?

**Comment by KowalskiP** [comment]
Yes, 3.5.1 works fine

**Comment by KowalskiP** [comment]
@cruzach hi
Any news about this?

**Comment by cruzach** [comment]
Not yet- it's not as easy as just updating a dependency since we have to make sure that works in all cases and doesn't have any side-effects we don't want

**Comment by KowalskiP** [comment]
Okay, it makes sense, thank you. I am waiting for your update =)

**Comment by christocracy** [comment]
[CocoaLumberjack @ 3.2.1](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/CHANGELOG.md#321---xcode-9-beta-on-aug-21st-2017) dates back to `2017-08-21`, over *two years* ago.

[3.5.0](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/CHANGELOG.md#350---xcode-10-on-jan-25th-2019) was built for XCode 10 and you *want* that ASAP.  Soon we'll be on XCode 11.

**Comment by Luckygirlllll** [comment]
@KowalskiP Is there any updates about this issue? Does your project has any side effects after updating CocoaLumberjack? Did you create the fork for ExpoKit version 34.0.0 ?

**Comment by Luckygirlllll** [comment]
@cruzach when will be this issue fixed in expoKit? I saw that [commit with new version of CocoaLumberjack ](https://github.com/expo/expo/pull/5500/commits/1fb47d06cd5867689147b98f69c3eb02fc346fc)was added 10 days ago, but it's not in ExpoKit 34.0.2. Will it be available in ExpoKit 34.0.3?

**Comment by cruzach** [comment]
Those update will probably go out with SDK 35

**Comment by Luckygirlllll** [comment]
@cruzach How can I use [this PR](https://github.com/expo/expo/pull/5500) in my ExpoKit project now? Should I update version of expoKit? or expo? What should be my steps?

**Comment by christocracy** [comment]
@Luckygirlllll just use [patch-package](https://github.com/ds300/patch-package) for now.

**Comment by Luckygirlllll** [comment]
@Luckygirlllll in `node_modules` folder I have `expo` and` expoKit` folder and I don't see in any of these folders `ExpoKit.podspec` file, as far as I understood in order to use `patch-package` I need to change  `ExpoKit.podspec` inside `node_modules`?

**Comment by duckbrain** [comment]
@Luckygirlllll I didn't try patch-package, but I don't think that will work because Cocoapods is pulling the dependency. We had to create a fork and and patch it ourselves. This points to our fork for Expo 34.

```ruby
  pod 'ExpoKit',
    :git => "http://github.com/ec2-software/expo.git",
    :tag => "ios/2.12.4-ec2-1",
```

Note `ios/2.12.4-ec2-1` is a patched version of `ios/2.12.3`, not `ios/2.12.4`, which has since been released. This is because Cocoapods wouldn't update it. I think it thinks the `-ec2` is pre-release and comes before it or something, but I am not very familiar with Cocoapods.

**Comment by Luckygirlllll** [comment]
@duckbrain Thanks, your solution works!

**Comment by pencilcheck** [comment]
> @Luckygirlllll I didn't try patch-package, but I don't think that will work because Cocoapods is pulling the dependency. We had to create a fork and and patch it ourselves. This points to our fork for Expo 34.
> 
> ```ruby
>   pod 'ExpoKit',
>     :git => "http://github.com/ec2-software/expo.git",
>     :tag => "ios/2.12.4-ec2-1",
> ```
> 
> Note `ios/2.12.4-ec2-1` is a patched version of `ios/2.12.3`, not `ios/2.12.4`, which has since been released. This is because Cocoapods wouldn't update it. I think it thinks the `-ec2` is pre-release and comes before it or something, but I am not very familiar with Cocoapods.

ios/2.12.5 still doesn't have this issue fixed, using your patched expokit works for me... this is really frustrating whenever we need to upgrade with expo.

**Comment by EvanBacon** [comment]
Closing as ExpoKit is now [deprecated in SDK 37](https://blog.expo.io/expo-sdk-37-is-now-available-dd5770f066a6). Please use the bare workflow instead, upgrading should now be a lot less frustrating.
